tinytodo/src/context.rs

```diff
@@ -511,14 +512,41 @@ impl AppContext {
     }
 
     fn get_lists(&self, r: GetLists) -> Result<AppResponse> {
-        let t: EntityTypeName = "List".parse().unwrap();
-        self.is_authorized(&r.uid, &*ACTION_GET_LISTS, &*APPLICATION_TINY_TODO)?;
+        let resource_ty: EntityTypeName = "List".parse().unwrap();
+        let entities: Entities = self.entities.as_entities(&self.schema);
+        let partial_request = RequestBuilder::default()
+            .action(Some(ACTION_GET_LIST.as_ref().clone().into()))
+            .principal(Some(cedar_policy::EntityUid::from(EntityUid::from(
+                r.uid.clone(),
+            ))))
+            .build();
+        let partial_response =
+            self.authorizer
+                .is_authorized_partial(&partial_request, &self.policies, &entities);
+
+        let slice = self
+            .entities
+            .euids()
+            .filter(|euid| euid.type_name() == &resource_ty);
 
         Ok(AppResponse::Lists(
-            self.entities
-                .euids()
-                .filter(|euid| euid.type_name() == &t)
-                .filter(|euid| self.is_authorized(&r.uid, &*ACTION_GET_LIST, euid).is_ok())
+                slice
+                .filter(|euid| matches!(
+                    partial_response.reauthorize(
+                        HashMap::from_iter(
+                            std::iter::once(
+                                ("resource".into(), RestrictedExpression::new_entity_uid(cedar_policy::EntityUid::from((*euid).clone())))
+                            )
+                        ),
```

Comment:
I believe this rerunning authorization for _every_ `List` entity in the entity store on the residual (substituting the entity for `resource`). Assuming I'm right, why is that an improvement over the old code? The old code basically did the same thing, but for the full policy set. But there are so few policies I'm not seeing this as much of an advantage.

Also: Have you confirmed this works as expected when all `policies.cedar` are available (none commented out)? I wonder/worry about implicit dependencies on the choice of policies in the policy set.